src/runtime/runtime_unix.go

```diff
@@ -307,3 +326,164 @@ func growHeap() bool {
 	setHeapEnd(heapStart + heapSize)
 	return true
 }
+
+func init() {
+	// Set up a channel to receive signals into.
+	signalChan = make(chan uint32, 1)
+}
+
+var signalChan chan uint32
+
+// Simple boolean that's true when any signals have been registered.
+var hasSignals uint32
+
+// Mask of signals that have been received. The signal handler atomically ORs
+// signals into this value.
+var receivedSignals uint32
+
+//go:linkname signal_enable os/signal.signal_enable
+func signal_enable(s uint32) {
+	if s >= 32 {
+		// TODO: to support higher signal numbers, we need to turn
+		// receivedSignals into a uint32 array.
```

Comment:
If we want to do it this way uint64 might even be sufficient.
checking the values for `SIGRTMAX` on my system leaves me with `64`. From a quick online look up is seems to be the case for many linux systems.

Reply:
Unfortunately, uint64 is not atomic on some systems such as MIPS32.
But signal numbers higher than 31 are rarely used (I think), so we can leave that for now and implement it when someone needs it.